### PR TITLE
pkp/pkp-lib#3578: harmonizing Form::fetch() signatures

### PR DESCRIPTION
**Pull request by crism**
Fixes part of #3578. May break some plugins.

### CODE REVIEW

File: classes/controllers/tab/settings/form/ContextSettingsForm.inc.php [hunk 1]
```diff
@@ -99,10 +99,9 @@ function readInputData($request) {
 
 	/**
 	 * @see Form::fetch()
-	 * @param $request PKPRequest
 	 * @param $params array optional
 	 */
-	function fetch($request, $params = null) {
+	function fetch($request, $template = null, $display = false, $params = null) {
```
Comment (asmecher) [inline classes/controllers/tab/settings/form/ContextSettingsForm.inc.php]: @crism, did you check for cases where `$display` is `true`? I haven't seen any of these for a while, and would rather remove them if it doesn't tangle things up too much.

Comment (crism) [reply classes/controllers/tab/settings/form/ContextSettingsForm.inc.php]: I’m not aware of any cases where either the template or the display are actually passed explicitly… but this API is used by themes and plugins, so I didn’t want to disrupt it, just do the minimum to end the error messages.

Comment (asmecher) [reply classes/controllers/tab/settings/form/ContextSettingsForm.inc.php]: I think that parameter was more a quirk of the way `TemplateManager::display` used to delegate to `TemplateManager::fetch` with a similar `$display` parameter. That got removed with the move to Smarty3, IIRC, so I'd recommend removing it here too. I don't think it's used and I find it weird and confusing. (Which is a common reaction for me, so YMMV.)

Comment (asmecher) [reply classes/controllers/tab/settings/form/ContextSettingsForm.inc.php]: This looks ready to go per the testing, but I'm not sure whether you saw the above comment.

**Pull request merged**